### PR TITLE
Space Traveler - Add fetch feature for missions

### PR DESCRIPTION
**Pull request by guerreiropedr0**
In this pull request, I:
- Installed redux, react-redux, redux-thunk dependencies
- Added fetch feature for missions

Here is my PR for the https://github.com/clintonjosephs/Space-traveler/issues/18 issue.

### CODE REVIEW

File: src/redux/ConfigureStore.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { applyMiddleware, createStore } from 'redux';
+import thunk from 'redux-thunk';
+import logger from 'redux-logger';
+import fetchMissions, { missionReducer } from './missions/missions';
+
+const store = createStore(missionReducer, applyMiddleware(logger, thunk));
+
+store.dispatch(fetchMissions());
+
```
Comment (clintonjosephs) [inline src/redux/ConfigureStore.js:9]: Great work so far 👏 👏 however, I have an observation here. Based on the requirements, please see image:

Screenshot 2022-02-28 at 20 51 36

The rockets component is the default view and therefore data should be fetched on the application's first run, however, for the mission's view **"the missions data should only be fetched (once) when a user navigates to the Missions section."** but based on the above code, missions data is fetched immediately store is initialized

Here are a few suggestions:
- Consider using the `useEffect` hook to fetch data when the user clicks to open the missions component
- The `useEffect` hook should not be passed an empty array dependency, rather the updated `missionState` array of objects can be passed as a dependency list `useEffect(() => {  use redux-thunk to fetch data }, [...state])` through a variable.

**Review by clintonjosephs — CHANGES_REQUESTED** [review]
Hey @guerreiropedr0,

You've done an amazing job so far, I'd like to highlight the following:
* Project is well structured
* Easy to read program statements
* Nice addition of error key to the mission state object.

However, I noticed something that can be refactored and your project will be complete. Kindly see below my comment:

File: src/redux/ConfigureStore.js (hunk 1 above)
Comment (guerreiropedr0) [reply src/redux/ConfigureStore.js:9]: Okay, I will do these changes tomorrow

**Review by clintonjosephs — APPROVED** [review]
Hey @guerreiropedr0,

<div>

You've done an amazing job so far, I'd like to highlight the following:
* Good work implementing the `useEffect` hook

Your project is complete! There is nothing else to say other than... it's time to merge it 

 Congratulations! 🎉
Cheers and Happy coding! 👏 👏 👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**